cluster/mesos/cluster.go

```diff
@@ -423,18 +424,30 @@ func (c *Cluster) scheduleTask(t *task) bool {
 	}
 
 	// Register the container immediately while waiting for a state refresh.
-	// Force a state refresh to pick up the newly created container.
-	// FIXME: unexport this method, see FIXME in engine.go
+
+	// In mesos 0.23+ the docker inspect will be sent back in the taskStatus.Data
+	// We can use this to find the right container.
+	inspect := []dockerclient.ContainerInfo{}
+	if data != nil && json.Unmarshal(data, &inspect) != nil && len(inspect) == 1 {
+		s.engine.RefreshContainer(inspect[0].Id, true)
```

Comment:
I can try, but it will add a lot of code as creating a `dockerclient.Container` from a `dockerclient.ContainerInfo` wil be done field by field